spec/models/course/condition/achievement_spec.rb

```diff
@@ -20,11 +22,29 @@
 
       context "when an achievement is already included in its conditional's conditions" do
         subject do
-          create(:course_condition_achievement, :duplicate_child, course: course).tap do |cca|
+          create(:course_condition_achievement, course: course).tap do |cca|
             allow(cca).to receive(:achievement_id_changed?).and_return(true)
           end
         end
-        it { is_expected.to_not be_valid }
+
+        it 'is not valid' do
+          expect(subject).to_not be_valid
+          expect(subject.errors[:achievement]).to_not be_blank
+        end
+      end
+
+      context 'when an achievement is required by another conditional with the same id' do
```

Comment:
Wait, hang on, is there a general condition table? I remember only specific conditional condition tables?